Chap_API_Publish.tex

```diff
@@ -195,25 +227,33 @@ \subsection{Range of Published Data}
+Published data and generated events are restricted to processes executing under the same instance of the host environment as the publisher or event creator.  Lookup of data is restricted to data published by processes running under the same instance of the host environment as the requester.
 %
 \declareconstitemvalue{PMIX_RANGE_LOCAL}{2}
-Data is only available to processes on the local node, or lookup is restricted to data published by processes on the local node of the requester.
+Published data and generated events are restricted to processes on the same node as the publisher or event creator.  Lookup of data is restricted to data published by processes on the same node as the requester.
 %
 \declareconstitemvalue{PMIX_RANGE_NAMESPACE}{3}
-Data is only available to processes in the same namespace, or lookup is restricted to data published by processes in the same namespace as the requester.
+Published data and generated events are restricted to processes in the same namespace as the publisher or event creator.
+Lookup of data is restricted to data published by procesess in the same namespace as the requester.
 %
 \declareconstitemvalue{PMIX_RANGE_SESSION}{4}
-Data is only available to all processes in the session, or lookup is restricted to data published by other processes in the same session as the requester.
+Published data and generated events are restricted to processes in the same session as the publisher or event creator.
+Lookup of data is restricted to data published by procesess in the same session as the requester.
 %
 \declareconstitemvalue{PMIX_RANGE_GLOBAL}{5}
-Data is available to all processes, or lookup is open to data published by anyone.
-%
-\declareconstitemvalue{PMIX_RANGE_CUSTOM}{6}
-Data is available only to processes as specified in the \refstruct{pmix_info_t} associated with this call, or lookup is restricted to data published by processes as specified in the \refstruct{pmix_info_t}.
+Published data and generated events are available to all processes within the domain of the host environment.
+Lookup of data is unrestricted and open to data published by any processes within the domain of the host enivornment as the requester.  This range differs from \refconst{PMIX_RANGE_RM} only on systems which have mechanisms to share events and
+publish/lookup data across multiple instances of a host environment.
 %
 \declareconstitemvalue{PMIX_RANGE_PROC_LOCAL}{7}
-Data is only available to this process, or lookup is restricted to data published by this process.
+Published data and generated events are available only to calling process.
+Lookup of data is restricted to data published by the calling process.
+%
+\declareconstitemvalue{PMIX_RANGE_CUSTOM}{6}
+Published data and generated events are restricted to processes
+described in the \refstruct{pmix_info_t} associated with this call.
+Lookup of data is restricted to data published by the processes described in
+in the \refstruct{pmix_info_t}.
```

Comment:
Note from 3Q 2022 (Aug. 11, 2022)
 - What does it mean for two `PMIX_RANGE_CUSTOM`s to match? Is it a strict ordering or is a subset fine?
 - Is the publisher required to be part of the range?
   - Publisher does need to be a member of the range?
   - "If I publish that data I own that data and can look it up so that I can unpublish"
     - Current description would prevent the lookup but not the unpublish.
     - Owner should always be able to look it up even if not in the range. Maybe need a clause here? What about overlapping keys?
 - Publish means "put some data for a set of consumers to lookup". The consumers might not know who published the data. For a custom range, those looking up might not know the full group. Alternatively, the lookup could specify a list including the publisher.

Reply:
Per IAWG on Aug. 15, 2022:
 * Does the publisher have to be part of the range it publishes to?
   * The publisher does not need to be a member of the range.
   * The publisher is the owner of the data, so it is the only one able to unpublish the data, but must specify the range where that data was published.
   * The publisher is the owner of the data but cannot lookup the data if it is not part of the range in which it was pushed.
 * If I'm looking up from a `PMIX_RANGE_CUSTOM` does it have to match what the publisher provided?
   * `PMIX_RANGE_NAMESPECE` (example)
     * **Publish**: means that the data is to be available to a set of lookup'ers in my namespace.
     * **Lookup**: means that I want data provided by a publisher in my namespace.
   * `PMIX_RANGE_CUSTOM`
     * **Publish**: 'custom' means I want the data to be available to this set of lookup'ers. Might not include the publisher.
     * **Lookup**: 'custom' means I want data provided by one of these publishers. Does not need to know all of the processes that have access to the data - only who is providing the data.
   * Generally `PMIX_RANGE_` means:
     * **Publish**: Define who can look up the data
     * **Lookup**: Define who published the data
 * Does the range have to match on the publish and lookup?
   * I'm publishing to range X, and I'm looking up on range X.
   * `PMIX_RANGE_CUSTOM` is the only one that takes an argument list of processes. It's the only one where the publisher and lookup'er might be using the same value associated with the `PMIX_RANGE_CUSTOM` key.
 * What if the publisher publishes the same key (different values) to different custom ranges that both include the client looking up the key? Or two different publishers both publish the same key (different values) that include the lookup client in different `PMIX_RANGE_CUSTOM`s.
   * This is not allowed.
   * The Server should reject the lookup operation as conflicting entries. The client should then change their custom range to include one
 * What If I want to lookup a value published by a process outside of my namespace?
   * Publisher will use `PMIX_RANGE_SESSION` or `PMIX_RANGE_CUSTOM` (if they know the lookup'er)
   * Lookup'er will user `PMIX_RANGE_CUSTOM` with the publisher's name
 * **Future work**: Maybe extend `PMIX_RANGE_NAMESPACE` with an argument of which namespace. Currently, it is the same as the caller.